types/underscore/index.d.ts

```diff
@@ -322,78 +323,34 @@ declare module _ {
         foldr: UnderscoreStatic['reduceRight'];
 
         /**
-        * Looks through each value in the list, returning the first one that passes a truth
-        * test (iterator). The function returns as soon as it finds an acceptable element,
-        * and doesn't traverse the entire list.
-        * @param list Searches for a value in this list.
-        * @param iterator Search iterator function for each element in `list`.
-        * @param context `this` object in `iterator`, optional.
-        * @return The first acceptable found element in `list`, if nothing is found undefined/null is returned.
-        **/
-        find<T>(
-            list: _.List<T>,
-            iterator: _.ListIterator<T, boolean>,
-            context?: any): T | undefined;
-
-        /**
-        * @see _.find
-        **/
-        find<T>(
-            object: _.Dictionary<T>,
-            iterator: _.ObjectIterator<T, boolean>,
-            context?: any): T | undefined;
-
-        /**
-        * @see _.find
-        **/
-        find<T, U extends {}>(
-            object: _.List<T> | _.Dictionary<T>,
-            iterator: U): T | undefined;
-
-        /**
-        * @see _.find
-        **/
-        find<T>(
-            object: _.List<T> | _.Dictionary<T>,
-            iterator: string): T | undefined;
-
-        /**
-        * @see _.find
-        **/
-        detect<T>(
-            list: _.List<T>,
-            iterator: _.ListIterator<T, boolean>,
-            context?: any): T | undefined;
-
-        /**
-        * @see _.find
-        **/
-        detect<T>(
-            object: _.Dictionary<T>,
-            iterator: _.ObjectIterator<T, boolean>,
-            context?: any): T | undefined;
-
-        /**
-        * @see _.find
-        **/
-        detect<T, U extends {}>(
-            object: _.List<T> | _.Dictionary<T>,
-            iterator: U): T | undefined;
+         * Looks through each value in the collection, returning the first one that passes a
+         * truth test (iteratee), or undefined if no value passes the test. The function
+         * returns as soon as it finds an acceptable element, and doesn't traverse the entire
+         * collection.
+         * @param collection Searches for a value in this collection.
+         * @param iteratee The truth test to apply.
+         * @param context `this` object in `iteratee`, optional.
+         * @return The first element in `collection` that passes the truth test or undefined
+         * if no elements pass.
+         **/
+        find<V extends Collection<any>>(
+            collection: V,
+            iteratee: Iteratee<V, boolean>,
```

Comment:
Something I forgot to mention in previous PRs: when the iteratee is transformed through `_.iteratee`, like in this case, the iteratee parameter is pretty much optional by definition since `undefined` turns into `_.identity`.

There are a few collection functions that don't transform the iteratee (right now I can think of `_.each` and `_.reduce`), but most do. Of the ones that do, there are a few where calling the function without an iteratee is pointless: `_.map` without an iteratee is just `_.toArray` and `_.sortBy` without an iteratee is just `Array.prototype.sort` with added overhead. In the vast majority of cases, however, including `find`, implicitly using `_.identity` by omitting the iteratee argument is both possible and meaningful.

```suggestion
            iteratee?: Iteratee<V, boolean>,
```

Reply:
Is there anything special that I should try to do in this case? I haven't experimented with it yet, but I feel like this would be expected to return the first truthy value; is it worth making a separate overload for this that removes `undefined` from the possible results rather than adding it?

Reply:
I think you don't need to do anything special. You are right that it will look for the first truthy value, but it might still return `undefined` if it doesn't find one.

Reply:
Good point, thanks!

Reply:
Made the update to find, reject, and filter as well.